specification/devops/resource-manager/Microsoft.DevOps/preview/2020-07-13-preview/devops.json

```diff
@@ -0,0 +1,920 @@
+        }
+      },
+      "required": [
+        "pipelineType",
+        "bootstrapConfiguration"
+      ]
+    },
+    "azurePipelineProperties": {
+      "type": "object",
+      "x-ms-discriminator-value": "azurePipeline",
+      "description": "Custom properties of a Azure Pipeline.",
+      "properties": {
+        "organization": {
+          "$ref": "#/definitions/OrganizationReference",
+          "description": "Reference to the Azure DevOps Organization containing the Pipeline. Required for 'azurePipeline' pipeline type."
+        },
+        "project": {
+          "$ref": "#/definitions/ProjectReference",
+          "description": "Reference to the Azure DevOps Project containing the Pipeline. Required for 'azurePipeline' pipeline type."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/PipelineProperties"
+        }
+      ],
+      "required": [
+        "organization",
+        "project"
+      ]
+    },
+    "githubWorkflowProperties": {
+      "type": "object",
+      "x-ms-discriminator-value": "githubWorkflow",
+      "description": "Custom properties of an Github Pipeline.",
+      "properties": {
+        "repository": {
+          "$ref": "#/definitions/CodeRepository",
+          "description": "Contains the specifications of the repository to be created for generating the workflow. Required for 'githubWorkflow' pipeline type."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/PipelineProperties"
+        }
+      ],
+      "required": [
+        "repository"
+      ]
+    },
+    "OrganizationReference": {
+      "type": "object",
+      "description": "Reference to an Azure DevOps Organization.",
+      "properties": {
+        "id": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Unique immutable identifier for the Azure DevOps Organization."
+        },
+        "name": {
+          "type": "string",
+          "description": "Name of the Azure DevOps Organization."
+        }
+      },
+      "required": [
+        "name"
+      ]
+    },
+    "ProjectReference": {
+      "type": "object",
+      "description": "Reference to an Azure DevOps Project.",
+      "properties": {
+        "id": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Unique immutable identifier of the Azure DevOps Project."
+        },
+        "name": {
+          "type": "string",
+          "description": "Name of the Azure DevOps Project."
+        }
+      },
+      "required": [
+        "name"
+      ]
+    },
+    "BootstrapConfiguration": {
+      "type": "object",
+      "description": "Configuration used to bootstrap a Pipeline.",
+      "properties": {
+        "sourceRepository": {
+          "$ref": "#/definitions/CodeRepository",
+          "description": "Repository containing the source code for the pipeline. Currently only 'azurePipeline' pipeline type supports this."
```

Comment:
@suyash691 is there a way to enforce this using discriminator?

Reply:
I do not have a pipelineType or any property to switch between types from hence this will have to exist as is. It doesn't make sense from an OO model that we use sourceRepository as a descriminator